File: fun-stuff/asciiArt.cpp
// Puzzle on CodinGame.com
// https://www.codingame.com/training/easy/ascii-art

#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

int main() {
  int letterWidth, letterHeight;
  std::string inputText;
  std::string alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";

  std::vector<std::vector<std::string>> letterTable;

  std::cin >> letterWidth;
  std::cin >> letterHeight;
  std::cin.ignore();
  getline(std::cin, inputText);

  std::transform(inputText.begin(), inputText.end(), inputText.begin(), ::toupper);

  letterTable.resize(letterHeight, std::vector<std::string>(27));

  for (int i = 0; i < letterHeight; i++) {
    int letterOffset = 0;
    std::string row;
    getline(std::cin, row);

    for (int j = 0; j < 27; j++) {
      letterTable[i][j] = row.substr(letterOffset, letterWidth);
      letterOffset += letterWidth;
    }
  }
  
  for (int i = 0; i < letterHeight; i++) {
    for (int j = 0; j < inputText.length(); j++) {
      if (alphabet.find(inputText[j]) != std::string::npos) {
        int letterIndex;
        letterIndex = alphabet.find(inputText[j]);
        std::cout << letterTable[i][letterIndex];
      } else {
        std::cout << letterTable[i][26];
      }
    }
    std::cout << std::endl;
  }

  return 0;
}

File: fun-stuff/asciiArt.py
# Puzzle on CodinGame.com
# https://www.codingame.com/training/easy/ascii-art

letter_width = int(input())
letter_height = int(input())
input_text = input().upper()
alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"

letter_table = [['' for _ in range(27)] for _ in range(letter_height)]

for i in range(letter_height):
    letter_offset = 0
    row = input()
    for j in range(27):
        letter_table[i][j] = row[letter_offset: letter_offset + letter_width]
        letter_offset += letter_width

for i in range(letter_height):
    for j in range(len(input_text)):
        if input_text[j] in alphabet:
            letter_index = alphabet.index(input_text[j])
            print(letter_table[i][letter_index], end='')
        else:
            print(letter_table[i][26], end='')
    print()
